presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DesugarRowSubscript.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.DesugarRowSubscriptRewriter;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class DesugarRowSubscript
+        extends ExpressionRewriteRuleSet
+{
+    public DesugarRowSubscript(Metadata metadata, SqlParser sqlParser)
+    {
+        super(createRewrite(metadata, sqlParser));
+    }
+
+    @Override
+    public Set<Rule<?>> rules()
```

Comment:
Any reason we remove `applyExpressionRewrite` from `ExpressionRewriteRuleSet:: rules()`?

Reply:
yes this results in a failure during rewriting an ExistsPredicate, please check the same in DesugarAtTimeZone

Reply:
We cannot analyze correlated subquery in `ExpressionAnalyzer` without the context of the whole query. Similar problem to analyzing lambda argument outside the context of function.